tests/unit/test_utils.py

```diff
@@ -1223,6 +1225,29 @@ def test_path_to_url_win():
     assert path_to_url('file') == 'file:' + urllib_request.pathname2url(path)
 
 
+@pytest.mark.parametrize('netloc, has_port, url', [
+    # Test domain name.
+    ('example.com', False, 'https://example.com'),
+    ('example.com:5000', True, 'https://example.com:5000'),
+    # Test IPv4 address.
+    ('127.0.0.1', False, 'https://127.0.0.1'),
+    ('127.0.0.1:5000', True, 'https://127.0.0.1:5000'),
+    # Test bare IPv6 address.
+    ('2001:DB6::1', False, 'https://[2001:DB6::1]'),
+    # Test IPv6 with port.
+    ('[2001:DB6::1]:5000', True, 'https://[2001:DB6::1]:5000'),
+    # Test IP with auth.
+    (
+        'user:password@localhost:5000',
+        True,
+        'https://user:password@localhost:5000'
+    )
+])
+def test_netloc_with_port(netloc, has_port, url):
```

Comment:
I would name this `test_build_url_from_netloc_and_netloc_has_port()` since you're testing both functions.